app/scripts/controllers/mmi-controller.js

```diff
@@ -549,22 +549,25 @@ export default class MMIController extends EventEmitter {
       this.preferencesController.setSelectedAddress(address);
     }
     const selectedChainId = parseInt(
-      this.networkController.getCurrentChainId(),
+      this.networkController.state.providerConfig.chainId,
       16,
     );
     if (selectedChainId !== chainId && chainId === 1) {
-      this.networkController.setProviderType('mainnet');
+      await this.networkController.setProviderType('mainnet');
     } else if (selectedChainId !== chainId) {
-      const network = this.preferencesController
-        .getFrequentRpcListDetail()
-        .find((item) => parseInt(item.chainId, 16) === chainId);
-      this.networkController.setRpcTarget(
-        network.rpcUrl,
-        network.chainId,
-        network.ticker,
-        network.nickname,
-      );
+      const foundNetworkConfiguration = Object.values(
```

Comment:
I'm not sure why this didn't cause issues before, but we must have missed this when migrating the custom networks list to NetworkController.

Reply:
Looks like this was introduced after that change, but was in PR for a long time. Probably a merge conflict.

Reply:
Ah, that makes sense.